src/utils/login.ts

```diff
@@ -0,0 +1,76 @@
+import User from '@services/User/User.service';
+import jwt_decode from 'jwt-decode';
+import { JwtPayload } from 'jwt-decode';
+const LOCAL_STORAGE_JWT_KEY = "Authorization";
+
+
+let invokeOnSignChange: { (): void }[] = [];
+export let onSignChange = (f: { (): void }) => {
+    invokeOnSignChange.push(f);
+};
+
+
+interface Jwt extends JwtPayload {
+    user_id: number;
+    is_admin: boolean;
+}
+export const signOut = () => {
+    localStorage.removeItem(LOCAL_STORAGE_JWT_KEY);
+    invokeOnSignChange.forEach(f => f());
+
+}
+export const signIn = async (
+    emailValue: string,
+    passwordValue: string,
+) => {
+    let token: string;
+    try {
+        token = await User.login(emailValue, passwordValue);
+        console.log({ token });
+    } catch (error) {
+        return false
```

Comment:
Now I see the reason behind those if's in those cases throwing an Error is better practice. For this implementation it doesn't matter, so you can leave it be